docs/lsm/partial_compiles/partial_compiles.rst

```diff
@@ -0,0 +1,141 @@
+        "NullResource(name=r1-eth0)" -> "NullResource(name=r1)"
+        "LifecycleTransfer(id=2)" -> "NullResource(name=r1-eth0)"
+
+        label = "ResourceSet for Router r1(id=0)";
+        labelloc = "top";
+        color = "green";
+    }
+    subgraph cluster_services_2 {
+        "NullResource(name=r2)" [shape=rectangle];
+        "LifecycleTransfer(id=3)";
+        "LifecycleTransfer(id=3)" ->  "NullResource(name=r2)" 
+
+        "NullResource(name=r2-eth0)" [shape=rectangle];
+        "NullResource(name=r2-eth0)" -> "NullResource(name=r2)"
+        "LifecycleTransfer(id=4)" -> "NullResource(name=r2-eth0)"
+
+        label = "ResourceSet for Router r2(id=3)";
+        labelloc = "top";
+        color = "green";
+    }
+
+
+In addition to the ``ResourceSets`` used by individual services, there are also ``Resources`` that are not in any set.
+These ``Resources`` can be shared by multiple services, with the limitation that any compile that produces them, has to produce them exactly the same. 
+For more information see :ref:`Partial Compiles<partial_compile>`.
+
+Service Instance Selection
+++++++++++++++++++++++++++
+
+To have efficiency gains when recompiling, it is important to only build the model for all Service Instances that are in the ``ResourceSet`` we want to update and nothing else. 
+
+This selection is done automatically within ``lsm::all``, based on the relations set between the service bindings as explained above. 
+
+The underlying mechanism is that when we recompile for a state change on any Service Instance, we first search its owner by traversing :inmanta:attribute:`lsm::ServiceEntityBinding.relation_to_owner` until we reach a single owner. 
+Then we traverse back down the :inmanta:attribute:`lsm::ServiceEntityBinding.relation_to_owner` until we have all children.
+``lsm::all`` will only return these children and nothing else.
+
+Limitations
+-------------------
+
+1. When doing normal compiles, the model can very effectively find conflicts between services (e.g. using indexes), because it has an overview of all instances. 
+When using partial compile, conflicts between groups can not be detected, because the compiler never sees them together. 
+This means that the model must be designed to be conflict free or rely on an (external) inventory to avoid conflicts. 
+*This is why we always advice to run models in full compile mode until performance becomes an issue*: it gives the model time to mature and to detect subtle conflicts.
+2. Complex topologies (with multiple parents or cross-relations) are currently not supported out of the box. 
+However, complex interdependencies between service instances are often an operation risk as well. 
+Overly entangled services are hard to reason about, debug and fix. 
+While it is possible to develop more complex topologies using the guidelines set out in :ref:`Partial Compiles<partial_compile>`, it may be preferable to simplify the service design for less interdependence. 
```

Comment:
Otherwise the itemize doesn't render correctly.
```suggestion
1. When doing normal compiles, the model can very effectively find conflicts between services (e.g. using indexes), because it has an overview of all instances. 
    When using partial compile, conflicts between groups can not be detected, because the compiler never sees them together. 
    This means that the model must be designed to be conflict free or rely on an (external) inventory to avoid conflicts. 
    *This is why we always advice to run models in full compile mode until performance becomes an issue*: it gives the model time to mature and to detect subtle conflicts.
2. Complex topologies (with multiple parents or cross-relations) are currently not supported out of the box. 
    However, complex interdependencies between service instances are often an operation risk as well. 
    Overly entangled services are hard to reason about, debug and fix. 
    While it is possible to develop more complex topologies using the guidelines set out in :ref:`Partial Compiles<partial_compile>`, it may be preferable to simplify the service design for less interdependence. 
```